### PR TITLE
Some basic tests for students#sped_referral Issue #8

### PR DESCRIPTION
**Pull request by eudaimonious**
not the best tests in the world but it's something...

### CODE REVIEW

**Comment by alexsoble** [comment]
@eudaimonious Basic tests are a wonderful start! And covering authorization first is a good idea.

File: spec/controllers/students_controller_spec.rb [hunk 1]
```diff
@@ -541,4 +541,50 @@ def make_request(query)
...
+        make_request({ student_id: student.id, format: :pdf })
+      end
+
+      context 'educator has schoolwide access' do
+
+        it 'is successful' do
+          expect(response).to be_success
+        end
+
+        it 'assigns the student correctly' do
```
Comment (erose) [inline spec/controllers/students_controller_spec.rb]: What is `assigns`?

File: spec/controllers/students_controller_spec.rb [hunk 2]
```diff
@@ -541,4 +541,50 @@ def make_request(query)
...
   end
 
+  describe '#sped_referral' do
+    let(:educator) { FactoryGirl.create(:educator, :admin, school: school) }
+    let(:school) { FactoryGirl.create(:school) }
+    let(:service) { FactoryGirl.create(:service, student: student) }
+    let(:student) { FactoryGirl.create(:student, :with_risk_level, school: school) }
+    let(:student_school_year) { FactoryGirl.create(:student_school_year, student: student) }
+
+    def make_request(options = { student_id: nil, format: :pdf })
```
Comment (erose) [inline spec/controllers/students_controller_spec.rb]: So in all these tests, we're making requests to the endpoint, then checking that the response contains something, right?

**Comment by erose** [comment]
I have some code questions that are purely for my own edification. You should feel free to merge this at any time.

**Comment by erose** [comment]
Ok I'm merging it cuz it rules.

File: spec/controllers/students_controller_spec.rb (hunk 1 above)
Comment (eudaimonious) [inline spec/controllers/students_controller_spec.rb]: I'm just using this to mean that the data stored in the instance variable matches what I expect based on the data in the factories. Not a great way to test honestly but meh.

File: spec/controllers/students_controller_spec.rb (hunk 2 above)
Comment (eudaimonious) [inline spec/controllers/students_controller_spec.rb]: I'm not familiar with RSpec. I copy and pasted my way into this. I'm not confident that RSpec actually makes real HTTP  requests. It's probably just a simulation. To my understanding, these tests will fake a request and check the values of the instance variables passed to the template.

**Comment by eudaimonious** [comment]
Sorry I was slow in responding. I'm all for edification!

**Pull request merged**